src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -83,7 +83,7 @@
         RegexLexer("atsign_literal", r"@[a-zA-Z_][\w]*", CodeSegment),
         RegexLexer("dollar_literal", r"[$][a-zA-Z0-9_.]*", CodeSegment),
     ],
-    before="not_equal",
+    before="like_operator",
```

Comment:
not_equal segment is now a multi-character segment rather than a lexer so just inserting before the next item down here 👍